src/session.ts

```diff
@@ -60,13 +60,16 @@ export class DebugSession implements IDebugger.ISession {
       this._client.iopubMessage.disconnect(this._handleEvent, this);
     }
 
-    this._client = client;
-
     if (client) {
+      this._client = client;
```

Comment:
Should this line be moved? I think it was left outside the `if` to be able to set the client to `null`.